src/compilation/download_packages.sh

```diff
@@ -5,11 +5,10 @@ script_dir=$(dirname "$0")
 . "$script_dir/utils.sh"
 
 # List of package URLs to download
-PACKAGE_URLS=(
+SOURCE_URLS=(
```

Comment:
you do not want to use your python script instead?

Reply:
Not yet, we should not use a hybrid bash/python build system.
If/When I finish converting build.sh to Python I'll open a separate MR.